doc/ifa-hld.md

```diff
@@ -0,0 +1,272 @@
+
+1.1 IFA feature is accomplished by configuring IFA session on various nodes that act as ingress,
+intermediate and egress devices. Device role is per flow in a node and a single node can act as ingress
+device for one flow and intermediate device for another flow.
+
+2.0.0.1 TAM device identifier to uniquely identify a device in network and insert the same in INT header.
+2.0.0.2 ACL configuration to identify a flow and sample packets from that flow to insert IFA headers.
+2.0.0.3 TAM collector configuration that can be attached to IFA flow on egress device to forward telemetry
+data.
+3.0 UI commands available to configure TAM device identifier, TAM collector and IFA configuration.
+3.1 UI commands available to show TAM device identifier, TAM collector, IFA configuration
+3.2 UI commands available to clear IFA configuration
+4.0 The maximum number of IFA 􀃖ows are platform dependent.
+4.1 Only one collector can be con􀃕gured in a device.
+4.5 Some platforms may require provisioning to enable IFA. 'ifa -config -enable' command to be issued to
+provision such platforms for IFA functionality. 'ifa -config -disable' command can be issued to disable
+provisioning of IFA on such platforms.
+
+### 1.1.2 Configuration and Management Requirements
+1. CLI configuration/show support
+2. REST API support
+3. gNMI support
+
+## 1.2 Design Overview
+### 1.2.1 Basic Approach
+
+As there is no opeconfig/ietf yang file exists for this feature, it is decided to go with sonic yang. 
+
+### 1.2.2 Container
+The changes are in sonic-management-framework container.
+
+There will be additional files added.
+1. XML file for the CLI
+2. Python script to handle CLI request (actioner)
+3. Jinja template to render CLI output (renderer)
+4. YANG models 
+	sonic-ifa.yang
+
+### 1.2.3 SAI Overview
+N/A
+
+# 2 Functionality
+## 2.1 Target Deployment Use Cases
+Whenever operator want to track packet latency/congestion metrics. 
+## 2.2 Functional Description
+The management UI provides an user friendly interface to configure Inband Flow Analyzer.
+# 3 Design
+## 3.1 Overview
+The packet action field in acl rule indicates the type of IFA device: ingress and egress. 'int_insert' makes the device as ingress IAF device and 'int_delete' as egress IFA device. Ingress IAF device makes a sample of a flow and tags them for analysis and data collection. Egress device is responsible for terminating IFA flow by summarizing the telemtry data of the entire path and sending it to the collector.
+
+The device identifer uniquely identifies the device in the network and inserts the ID in the IFA header.
+
+Collectors receive the telemetry data from egress devices.
+
+Flow configuration will contain sampling rate at which rate traffic will be sampled.
+
+IFA feature can be enabled or disabled.
+## 3.6 User Interface
+### 3.6.1 Data Models
+
+https://github.com/project-arlo/sonic-mgmt-framework/blob/94a7c70de961c5c9b59429bc4acdba710a2592b8/models/yang/sonic/sonic-ifa.yang
```

Comment:
@spenugondaa  : I think common TAM configurations like device-id, collector should be there in common sonic-tam yang file as those configurations are common for multiple features. Only feature specific configurations should be present in feature specific YANG file.  TAM specific configurations should in different XML file for CLI also .